src/core/server/http/router/validator/validator.ts

```diff
@@ -0,0 +1,280 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ValidationError, Type, schema, ObjectType } from '@kbn/config-schema';
+import { Stream } from 'stream';
+import { RouteValidationError } from './validator_error';
+
+/**
+ * Validation resolver to be used in the custom validation function
+ *
+ * See {@link RouteValidationFunction}.
+ *
+ * @public
+ */
+export interface RouteValidationResolver {
```

Comment:
nit: can we call this `RouteValidationResultFactory` to closer match the `KibanaResponseFactory` name?

Reply:
I've also changed the interface so it returns the methods `ok` and `badRequest` (instead of `fail`) to be an even closer match to the response factory :)